### PR TITLE
2263 splitting LFRic scalar args

### LINKED ISSUE #2263: Splitting LFRicScalarArgs from src/psyclone/dynamo0p3.py
Sub-issue of #2235

### PR DESCRIPTION
**Pull request by oakleybrunt**
closes #2263

### CODE REVIEW

**Review by TeranIvy — CHANGES_REQUESTED** [review]
Nice work! There are some minor comments and clean-ups (see inline).

Tests with compilation work and examples pass. I didn't check documentation in this instance, but grepping for `LFRicScalarArgs` only returns source and test changes so I think we're fine here.

Also, the branch needs updating with master. I'd advise doing that in a separate commit to review corrections.

File: src/psyclone/dynamo0p3.py [hunk 1]
```diff
@@ -61,7 +61,7 @@
                                    LFRicArgDescriptor, KernelInterface,
                                    LFRicCollection, LFRicConstants,
                                    LFRicSymbolTable, LFRicInvoke,
-                                   LFRicKernCallFactory)
+                                   LFRicKernCallFactory, LFRicScalarArgs)
```
[the remark this answers is not in the document]
Comment (oakleybrunt) [reply src/psyclone/dynamo0p3.py:64]: Hmm, that's strange.

`Counter` is used on line 134 and `FORTRAN_INTENT_NAMES` is used on line 76.

My `pylint` shows this:

```
************* Module psyclone.domain.lfric.lfric_scalar_args
domain/lfric/lfric_scalar_args.py:68:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
domain/lfric/lfric_scalar_args.py:182:4: R0914: Too many local variables (16/15) (too-many-locals)
domain/lfric/lfric_scalar_args.py:182:4: R0912: Too many branches (22/12) (too-many-branches)
```

**Review by TeranIvy — COMMENTED** [review]
All review comments are addressed. Tests with compilation pass except for an unrelated test. It is not the fault of this PR and it is exposed by using GCC 11.2.0 compiler which is stricter in syntax. 
Documentation builds fine. Examples run.
After updating the user guide and changelog I will proceed to merge the branch.

**Review by TeranIvy — APPROVED** [review]
CI passed, proceeding to merge.

**Pull request merged**